### PR TITLE
Upgrade to react-map-gl v7

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/9706

_Note: This will not be merged until drawing components are updated to use `@mapbox/mapbox-gl-draw` instead of `react-map-gl-draw`. PR to come, but I wanted to try to split this up as much as possible._

Summary of changes:
1. Upgraded from `react-map-gl ^5.2.10` to `react-map-gl ^7.0.18`
2. Removed `react-map-gl-draw` and `react-map-gl-geocoder` (not compatible with react-map-gl v7)
3. Replaced`react-map-gl-geocoder` with `@mapbox/mapbox-gl-geocoder` (this Mapbox plugin is the way forward 📍 🙌 )
4. Removed the portal that added the speed dial and made it a plain component to reduce the complexity there a bit
5. Updated styling on the map controls to align with Mapbox styling - I can revert this but it was screaming at me to tighten up 😅 and was a small effort (box-shadow, spacing, and borders)

## Testing
**URL to test:** 
https://deploy-preview-753--atd-moped-main.netlify.app/moped/

**Steps to test:**

Devs - please feel free to reach out to me for questions about the code diff. Lots of the code changes came from Prettier formatting so apologies about that.

1. Can switch layers with speed dial
2. Can zoom in/out
3. Can find **Barton Springs Pool** with the geocoder
4. Can add and edit a component that is a line (Bike Lane, etc.)
5. Can add and edit a component that is a point (Bike Box, etc.)
6. Can add and edit a component that is multiple lines (Bike Lane, etc.)
7. Can add and edit a component that is multiple points (Bike Box, etc.)
8. Can hover line segments and see tooltip with street label from CTN layer while selecting
9. Can hover line segments and see tooltip with street label from CTN layer while viewing existing project components
10. Map fits to project components if there are some selected on load in Project Summary and Map tabs
11. Map fits to components when selected for edit
12. Map Features dropdown can be used to show/hide lines and point in both view and edit modes

---
#### Ship list
- [x] Code reviewed 
- [ ] Product manager approved
- [x] ~Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~

### CODE REVIEW

File: moped-editor/src/components/Maps/GeocoderControl.js
Comment (mddilley) [inline moped-editor/src/components/Maps/GeocoderControl.js:20]: This entire file was copied and de-Typscriptified straight from the `react-map-gl` code example (with small mods like the bounding box to keep search results in the Austin Full Purpose Jurisdiction and to turn off showing a mark on the location selected).

File: moped-editor/src/components/Maps/MapToolsCollapse.js
Comment (mddilley) [inline moped-editor/src/components/Maps/MapToolsCollapse.js:45]: New component that came out of figuring out how the map component interface worked - not 100% necessary but it was helpful to wrangle some styles that had fallen out of sync in two separate files.

File: moped-editor/src/utils/mapHelpers.js
Comment (mddilley) [inline moped-editor/src/utils/mapHelpers.js:106]: Most of the pixel numbers came from preventing the geocoder replacement component from overlapping other map controls (they use 10px for spacing for their controls and specific widths for button, etc.). This could be tightened up, but that could be a backlog issue.

File: moped-editor/src/utils/mapHelpers.js
Comment (mddilley) [inline moped-editor/src/utils/mapHelpers.js:145]: The replacement geocoder can position itself so we don't need a wrapper anymore. Some of these style overrides are moved to a file called `ProjectComponentsMap.css`

File: moped-editor/src/utils/mapHelpers.js
Comment (mddilley) [inline moped-editor/src/utils/mapHelpers.js:713]: `srcEvent` is no longer exposed by `react-map-gl` so the offset has to come from `originalEvent` now

File: moped-editor/src/utils/mapHelpers.js
Comment (mddilley) [inline moped-editor/src/utils/mapHelpers.js:733]: A big update here is that `useFeatureCollectionToFitBounds` no longer handles setting and getting the viewport state. This is an improvement because we don't have to alter the viewport to update the bounds of the map. We can just use Mapbox's already existing [map.fitBounds()](https://docs.mapbox.com/mapbox-gl-js/example/fitbounds/) method.

Also separates the viewport state management from this custom hook which wasn't great in the first place (but a better choice before version 7).

File: moped-editor/src/views/projects/projectView/ProjectComponentsBaseMap.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponentsBaseMap.js:49]: Small improvement to keep the OOTB controls aligned with the speed dial

File: moped-editor/src/views/projects/projectView/ProjectComponentsBaseMap.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponentsBaseMap.js:89]: I converted this from a portal to a plain old component because I was having trouble piecing together the container the portal tied to, its ref, and this component. If anyone disagrees, I'll change it back.

File: moped-editor/src/views/projects/projectView/ProjectComponentsMap.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMap.js:251]: Cursor updates are handled differently in v7 and I used [this example from the docs](https://github.com/visgl/react-map-gl/tree/7.0-release/examples/custom-cursor)

File: moped-editor/src/views/projects/projectView/ProjectComponentsMap.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMap.js:350]: This will be removed in the next PR, but I kept it around so I could reference what the cursor does when drawing.

File: moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js:713]: This will be changed back to using the save reducer when drawing in upgraded in the next PR.

File: moped-editor/src/views/projects/projectView/ProjectComponentsMapView.js
Comment (chiaberry) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMapView.js:24]: thank you for including this

File: moped-editor/src/components/Maps/GeocoderControl.js
Comment (chiaberry) [inline moped-editor/src/components/Maps/GeocoderControl.js:130]: trying to wrap my head around how the geocoder control component works, but will this be uncommented after the draw PR is approved? or are these commented out because they don't apply to moped's map use case

Comment (mddilley) [reply moped-editor/src/components/Maps/GeocoderControl.js:130]: That is a good point - these are commented out in the react-map-gl example because their types are not covered in `@types/mapbox__mapbox-gl-geocoder` which doesn't apply to us since we aren't using TypeScript. I'll remove them since that is confusing. 👍

File: moped-editor/src/views/projects/projectView/ProjectComponentsBaseMap.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/ProjectComponentsBaseMap.js:89]: 👍 have never personally been able to get a portal to work, so i'm happy to see it go 🚀

File: moped-editor/src/views/projects/projectView/ProjectComponentsMap.js
Comment (johnclary) [reply moped-editor/src/views/projects/projectView/ProjectComponentsMap.js:250]: v nice!

File: moped-editor/src/components/Maps/MapToolsCollapse.js
Comment (johnclary) [reply moped-editor/src/components/Maps/MapToolsCollapse.js:45]: i like it!

**Review by johnclary — APPROVED** [review]
works as promised! the performance seems real good, even when rendering a bajillion streets.

thanks @mddilley!! really looking forward to the next iteration with draw.

File: moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js:770]: Forgot to note that this was to squash a console warning about nested `<h2>` tags (`DialogTitle` returns a `<h2>`)

**Comment by mateoclarke** [comment]
Seems to work fine locally, but the nearmap API doesn't work in the preview app so the speed dial layer switch you listed doesn't work. Assuming this is just a know whitelisting issue
Screen Shot 2022-08-25 at 2 08 59 AM
.

**Review by mateoclarke — APPROVED** [review]
🚢💨✨

**Review by frankhereford — APPROVED** [review]
Mike, this is a huge amount of work! Thank you so much for taking this on. 🚀

I've gone through all the testing steps you presented, and it worked great. I've read the diff on the PR, and that was very helpful - thank you. 

I left a comment that it makes sense in the upcoming db schema refactor to store the geometry be stored in WGS84. This will align with the projection that the map uses. Please let me know if there is a better default, and we can switch to it 🙏.

**Review by chiaberry — APPROVED** [review]

**Comment by chiaberry** [comment]
I noticed when I add a signal component the map doesnt recenter to fit the signal in the map bounds. I tested on staging and the map does recenter to include signal components. 

Imo this feels like a later bug fix / enhancement, so I'm keeping my PR approval ✅

**Comment by mddilley** [comment]
@mateoclarke @chiaberry Thanks for catching those bugs 🙏 , and I'll create backlog issues for:
- Need to check on if we can add wildcards to the NearMap approval list (have to go through CTM for this)
- Make map recenter to fit the signal in the map bounds for signal components

**Comment by amenity** [comment]
What are the additional layers that show up under "Map features" drop-down when editing? 
Screen Shot 2022-08-29 at 11 35 08 PM

**Review by amenity — APPROVED** [review]
Per ☝🏻 , I'm a bit confused about the "Map Features" dropdown menu... In fact, I'm not really certain what the purpose of that menu is for users. 🤔 

I'm going to go ahead and approve assuming that I'm just confused about a few things here.

**Pull request merged**
